### PR TITLE
Minor update based on new spec changes

### PR DESCRIPTION
**Pull request by souravpNV**

### CODE REVIEW

**Comment by souravpNV** [comment]
Adding @jeffbolznv  for review, thanks.

File: layers/core_validation.cpp [hunk 1]
```diff
@@ -12029,6 +12029,11 @@ bool CoreChecks::PreCallValidateGetRayTracingShaderGroupHandlesKHR(VkDevice devi
                          "VkPhysicalDeviceRayTracingPropertiesKHR::shaderGroupHandleSize.",
                          dataSize);
     }
+    if (firstGroup > pipeline_state->raytracingPipelineCI.groupCount) {
```
Comment (jeffbolznv) [inline layers/core_validation.cpp:12032]: Should be `>=`

File: layers/core_validation.cpp [hunk 2]
```diff
@@ -12048,6 +12053,11 @@ bool CoreChecks::PreCallValidateGetRayTracingCaptureReplayShaderGroupHandlesKHR(
                          dataSize);
     }
     const PIPELINE_STATE *pipeline_state = GetPipelineState(pipeline);
+    if (firstGroup > pipeline_state->raytracingPipelineCI.groupCount) {
```
Comment (jeffbolznv) [inline layers/core_validation.cpp:12056]: `>=` here too

File: layers/drawdispatch.cpp [hunk 3]
```diff
@@ -434,6 +434,30 @@ bool CoreChecks::PreCallValidateCmdTraceRaysNV(VkCommandBuffer commandBuffer, Vk
                                                uint32_t width, uint32_t height, uint32_t depth) const {
     bool skip = ValidateCmdDrawType(commandBuffer, true, VK_PIPELINE_BIND_POINT_RAY_TRACING_NV, CMD_TRACERAYSNV,
                                     "vkCmdTraceRaysNV()", VK_QUEUE_COMPUTE_BIT);
+    const CMD_BUFFER_STATE *cb_state = GetCBState(commandBuffer);
+    skip |= InsideRenderPass(cb_state, "vkCmdTraceRaysNV()", "VUID-vkCmdTraceRaysNV-renderpass");
+    auto callable_shader_buffer_state = (BUFFER_STATE *)GetBufferState(callableShaderBindingTableBuffer);
+    if (callableShaderBindingOffset > callable_shader_buffer_state->createInfo.size) {
```
Comment (jeffbolznv) [inline layers/drawdispatch.cpp:440]: `>=` for all these

File: layers/drawdispatch.cpp [hunk 4]
```diff
@@ -434,6 +434,30 @@ bool CoreChecks::PreCallValidateCmdTraceRaysNV(VkCommandBuffer commandBuffer, Vk
...
+        skip |= LogError(
+            commandBuffer, "VUID-vkCmdTraceRaysNV-callableShaderBindingOffset-02461",
+            "vkCmdTraceRaysNV: callableShaderBindingOffset must be less than the size of callableShaderBindingTableBuffer.");
```
Comment (jeffbolznv) [inline layers/drawdispatch.cpp:443]: Would be nice to print out the values for some of these.

File: layers/drawdispatch.cpp [hunk 5]
```diff
@@ -434,6 +434,30 @@ bool CoreChecks::PreCallValidateCmdTraceRaysNV(VkCommandBuffer commandBuffer, Vk
...
+    }
+    auto hit_shader_buffer_state = (BUFFER_STATE *)GetBufferState(hitShaderBindingTableBuffer);
+    if (hitShaderBindingOffset > hit_shader_buffer_state->createInfo.size) {
```
Comment (jeffbolznv) [inline layers/drawdispatch.cpp:446]: All of the buffers except for raygenShaderBindingTableBuffer are optional and these can be NULL.

File: layers/core_validation.cpp (hunk 1 above)
Comment (souravpNV) [reply layers/core_validation.cpp]: updated for now, I left this intentionally, should it be less than or equal to in the spec? same as for firstGroup + groupCount check below. 

updated rest all suggestions

File: layers/core_validation.cpp [hunk 6]
```diff
@@ -12048,6 +12053,11 @@ bool CoreChecks::PreCallValidateGetRayTracingCaptureReplayShaderGroupHandlesKHR(
                          dataSize);
```
Comment (jeffbolznv) [inline layers/core_validation.cpp:12053]: Can you fix the error message for this one?

File: layers/core_validation.cpp (hunk 1 above)
Comment (jeffbolznv) [reply layers/core_validation.cpp]: The rules should be that first must be less than the max, and first+count must be less than or equal to the max. So the code looks OK now.

File: layers/core_validation.cpp (hunk 6 above)
Comment (jeffbolznv) [inline layers/core_validation.cpp:12053]: Can you fix the error message for this one?

Comment (souravpNV) [reply layers/core_validation.cpp:12053]: Updated, not sure if you were pointing to this one.

VUID-vkGetRayTracingCaptureReplayShaderGroupHandlesKHR-firstGroup-03483
This was missing 'to' in the error message, rest two looked fine.

Comment (jeffbolznv) [reply layers/core_validation.cpp:12053]: Yeah, I meant this one (2419). It doesn't have a real string, just prints out datasize, which will not be helpful.

Comment (souravpNV) [reply layers/core_validation.cpp:12053]: I think that is an issue with how diff is being displayed, the string field is not empty.
This is 2419:
            device, "VUID-vkGetRayTracingShaderGroupHandlesKHR-firstGroup-02419",
            "vkGetRayTracingShaderGroupHandlesKHR: The sum of firstGroup and groupCount must be less than or equal the number "
            "of shader groups in pipeline.");

the datasize in line 12053  pertain to 

        skip |= LogError(device, "VUID-vkGetRayTracingCaptureReplayShaderGroupHandlesKHR-dataSize-03484",
                         "vkGetRayTracingCaptureReplayShaderGroupHandlesKHR: dataSize (%zu) must be at least "
                         "VkPhysicalDeviceRayTracingPropertiesKHR::shaderGroupHandleCaptureReplaySize.",
                         dataSize);

Comment (jeffbolznv) [reply layers/core_validation.cpp:12053]: Oh weird, even when I click the "expand" button it doesn't show the lines between 12039 and 12053. Nevermind, my bad.

Comment (souravpNV) [reply layers/core_validation.cpp:12053]: Thanks, Jeff, so this looks gtg?

Comment (jeffbolznv) [reply layers/core_validation.cpp:12053]: yep

**Pull request merged**